src/sentry/migrations/0729_add_subscription_fk_to_incident.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 5.0.6 on 2024-06-12 16:47
+
+import logging
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def _backfill_incident_subscription(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Incident = apps.get_model("sentry", "Incident")
+
+    # use RangeQuerySetWrapper to avoid loading all Incidents into memory
+    for incident in RangeQuerySetWrapper(Incident.objects.all()):
+        alert_rule = incident.alert_rule
+        if not alert_rule:
+            logger.info(
+                "Incident found with no alert_rule. skipping",
+                extra={"incident_id": incident.id},
+            )
+            continue
+
+        snuba_query = alert_rule.snuba_query
+        subscriptions = list(snuba_query.subscriptions.all())
+        if len(subscriptions) == 0 or len(subscriptions) > 1:
+            logger.info(
+                "Incident found with no or multiple subscriptions. skipping",
+                extra={
+                    "incident_id": incident.id,
+                    "snuba_query_id": snuba_query.id,
+                    "subscription_ids": [
+                        subscription.id for subscription in subscriptions if len(subscriptions) > 0
+                    ],
+                },
+            )
+            continue
```

Comment:
Do we not want to just always add the subscription to the incident? Or is the concern we don't know which one to add?

Reply:
Yea - don't know which one to add

for the moment i think this should be fine since the only alert rules with multiple subscirptions sould be our test rules